### DISCUSSION TITLE
BoM - is there a better way to get package size?

### DISCUSSION THREAD

**Discussion by tingox** [INTI-CMNB/KiBot#481] [Q&A]
When creating BOMs (either csv or html format, haven't tried others yet) you get the footprint value as it is in KiCad, example "C_0603_1608Metric", "R_0603_1608Metric", "LED_0603_1608Metric" and so on., If I want to get just the "0603" part, I can create a separate field (say, Package) in KiCad and populate it manually. This isn't ideal.
Is there a better way?

**Comment by set-soft** [comment]
Is this information somewhere? I don't think KiCad keeps this information.

I think you can use a _Package_ field and populate it using information from component distributors, like digi-key.
If you already have a field containing the distributor code, and I strongly recommend it to get a coherent BoM, use the *spec_to_field* filter to transfer the package information from the distributor to the _Package_ field.
This isn't simple, but the steps needed are the ones needed to get information about price and availability of the components, which  is very important for a real life BoM.

My suggestion: 
- Choose a distributor and add the distributor component code to each component
- Group your components using this code
- Make sure you can get price & availability data (enabling KiCost options in the BoM)
- Generate the _specs_ sheet (along with the price sheet).
- Inspect the _specs_ informed by your distributor and choose the one you want to use as package.
- Create a KiBot config to apply a filter that transfers it to the schematic and saves the schematic to a new file.
- Check the new schematic is what you want and replace the old one (use a revision system like git to be able to unroll changes!)

You'll get what you want and much more. Of course, setting up an account to get the data from a distributor is an annoying process, but it worth the effort.

**Comment by tingox** [comment]
I was thinking more along the lines of being able to transform the information already in KiCad (e.g. "R_0603_1608Metric") via a filter, function or something like that.

Your suggestion is reasonable, if my goal was to find available parts from one or more distributors. However, for this project I am only generating the BOM. Somebody else is tasked with finding and buying parts.

This particular project doesn't have too many parts, so I'll stick with the manually populated Package filed for now
